src/translator_pool.cc

```diff
@@ -32,40 +32,44 @@ namespace ctranslate2 {
     job.options = options;
 
     {
-      std::lock_guard<std::mutex> lock(_mutex);
+      std::unique_lock<std::mutex> lock(_mutex);
+      _can_add_more_work.wait(lock, [this]{ return _work.size() < 2 * _workers.size(); });
```

Comment:
I'm thinking that the `post` method should never block by default. There could be users of this class who don't care about memory and just want to post their job without wait time.

Could you look into making this optional, or move the wait condition into `consume_stream`?

Reply:
Ok